### DISCUSSION TITLE
How to show compile sql instead of raw sql in UI for dagster dbt integration?

### DISCUSSION THREAD

**Discussion by ZhiZinTan-KSO** [dagster-io/dagster#17223] [Q&A] [answered]
How to show compile sql instead of raw sql in UI for dagster dbt integration?
I've found an old discussion but is no longer supported in the new version (https://github.com/dagster-io/dagster/discussions/14477#discussioncomment-6065088).
![image](https://github.com/dagster-io/dagster/assets/107019210/7bc56bc7-a770-4b16-85d2-8e3cec969931)
image taken from above old discussion.

**Comment by rexledesma** [comment, marked as answer]
This is still supported in the current version, the API names are just different. As mentioned in the comments in that thread:

1. You will need to create your dbt manifest using `dbt compile`. This way, the compiled code will be present in the dbt manifest ([link](https://github.com/dagster-io/dagster/discussions/14477#discussioncomment-6075216)). 
2. Then, you can define a custom `DagsterDbtTranslator` with a `get_description` method that takes the `compiled_code` from the dbt manifest and sets it as the description for your dbt assets. The implementation should be similar to this [previous comment](https://github.com/dagster-io/dagster/discussions/14477#discussioncomment-6065088).

Some references:
- [Customizing descriptions of your dbt assets](https://docs.dagster.io/integrations/dbt/reference#customizing-descriptions)
- [API docs for `DagsterDbtTranslator.get_description`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DagsterDbtTranslator.get_description)
